web/models/setting/system_variable.ex

```diff
@@ -49,6 +50,16 @@ defmodule CoursePlanner.SystemVariable do
     |> validate_editable()
   end
 
+  def validate_by_record_required(changeset, target_params) do
```

Comment:
should we call this function instead of `validate_required` in the update changeset?